projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1330,49 +1341,49 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      * The custom template, if any, that should be used when rendering a row expand indicator.
      */
     @ContentChild(IgxRowExpandedIndicatorDirective, { read: TemplateRef })
-    public rowExpandedIndicatorTemplate: TemplateRef<any> = null;
+    public rowExpandedIndicatorTemplate: TemplateRef<IgxGridRowTemplateContext> = null;
```

Comment:
The row expanded/collapsed templates seem to be reused between different expansion templates.
In most places the context type is row, but in the expandable-cell (the one used for master-detail grid) the context is the cell:

![image](https://user-images.githubusercontent.com/10397980/190984424-e3460032-83e8-4db2-be98-03c3399586cb.png)

That one should probably be updated for consistency. Otherwise its type will mismatch.